### ISSUE TITLE
[Docs] Example for declarative config contains deprecated ``data_files``

### ISSUE THREAD

**Issue by abravalheri** [pypa/setuptools#2832]
### Summary

According to the following references:

- https://setuptools.pypa.io/en/latest/references/keywords.html
- https://packaging.python.org/guides/distributing-packages-using-setuptools/#package-data

the usage of ``data_files`` is deprecated and does not work with wheels.

Therefore I think the best is to not include it in the main example on the [declarative configuration docs](https://setuptools.pypa.io/en/latest/userguide/declarative_config.html) (since it may give the wrong impression/idea to users).

Instead the best should be directing them to the [datafiles support docs](https://setuptools.pypa.io/en/latest/userguide/datafiles.html)

### Additional Information

The example is particularly controversial, since it includes files to be written to the `/etc` folder.

![Example configuration](https://user-images.githubusercontent.com/320755/138883140-6c368eba-4367-4d2a-a214-588a3bb69d7e.png)

### Code of Conduct

- [X] I agree to follow the PSF Code of Conduct

**Comment by pelson** [comment]
Sorry to revive this >2 year old MR, but is ``data_files`` really deprecated? https://packaging.python.org/en/latest/guides/distributing-packages-using-setuptools/#data-files does not mention this (it only references eggs in terms of issues with ``data_files``).

The use-case described in that documentation remains valid IMO:

> Although configuring [package_data](https://packaging.python.org/en/latest/guides/distributing-packages-using-setuptools/#package-data) is sufficient for most needs, in some cases you may need to place data files outside of your [packages](https://packaging.python.org/en/latest/glossary/#term-Import-Package)

**Comment by abravalheri** [comment]
Hi @pelsom, this was already discussed in the Python discourse (I am on my phone roght now so it is hard for me to find the link, sorry for that). The conclusion was that there is a single ***partially*** valid use case for `data_files`: when you have a plugin system that needs to find asset files (e.g. js, css, html) from multiple plugins AND the amount of plugins is very very big.

I say ***partially valid***, because even in this use case `data_files` is not strictly necessary and a very good/performant replacement is to use
a *namespace package* as a container for the shared files. So we are really still missing a valid use case that can only be achieved with`data_files`.

Even in this one partially valid use case, it is impossible to fully select 100% where the files will end up, because that will depend on the installation schema chosen by the end user - not dev. For example, the installation root maybe relative to a virtual environment folder, the user'shome directory, whichever folder the user selectev via `pip install --target ...`, etc ... 

We could bring the documentation for `data_files` back, but that would require a completely different text and examples then this one that was removed. The example that was removed is an ***invalid*** use of `data_files` (any use of `data_files` trying to add files to `/etc` or `/usr` is invalid and likely to not work as intended due to the way Python works with virtual environments). The coice of removal was to avoid misleading examples.

If any one would like to work on a PR for documeting `data_files`, I think that a good start point is expading the documentation in the package data page. We should comment on the shortcommings of `data_files`, discuss on how all the other alternatives are more reliable and then talk about the only partially valid use case, still recomending the alternative of namespace packages instead.

I have in mind for ages the idea of flowchart for deciding which technique to ship non-Python files, but I never had the time 😜.

Anyway, let's say that `data_files` is not completely gone, but all the use cases (except the one I discussed) is no longer supported. Instead devs are encouraged to use other techniques.
